### ISSUE TITLE
ACF (Pro) has fixed its init call function since version 6.1.0 and thus no longer executes the call of the PHP function.

### ISSUE THREAD

**Issue by 3ele-projects** [AdvancedCustomFields/acf#804]
Versions: ACF Pro > 6.1.0

ACF has fixed its init call function since version 6.1.0 and thus no longer executes the call of the PHP function.

Normal example from ACF Pro:

if( function_exists('acf_add_local_field_group') ):

https://www.advancedcustomfields.com/resources/register-fields-via-php/

The solution is per 

add_action('acf/init', 'kcp_init');

function kcp_init(){

if( function_exists('acf_add_local_field_group') ):

to wait until ACF is initialised.

__

Please update your Manuals

https://www.advancedcustomfields.com/resources/register-fields-via-php/

**Comment by eyetractive** [comment]
I can confirm that something has changed. Our hooks added to "acf/init" are not working anymore, at all.

This is a bug? Or an intended, but undocumented change?!

**Comment by nicmare** [comment]
thanks for the report. this should to the trick right?

```
add_action("acf/init",function(){
if( function_exists('acf_add_local_field_group') ):
	
	acf_add_local_field_group(array(
		'key' => 'group_639ca3bf19def',
		'title' => 'Block: Content-Block',
		'fields' => array(
			array(
				'key' => 'field_639ca3bfbbaba',
				'label' => 'Content-Block',
				'name' => 'content-block',
				'aria-label' => '',
				'type' => 'select',
				'instructions' => 'Inhalte <a href="edit.php?post_type=ct_content_block" target="_blank">hier anlegen / ändern</a>',
				'required' => 0,
				'conditional_logic' => 0,
				'wrapper' => array(
					'width' => '',
					'class' => '',
					'id' => '',
				),
				'choices' => array(
					1701 => 'Wetter',
					1334 => 'Öffnungszeiten',
				),
				'default_value' => false,
				'return_format' => 'value',
				'multiple' => 0,
				'allow_null' => 0,
				'ui' => 0,
				'ajax' => 0,
				'placeholder' => '',
			),
		),
	));

endif;
});
```

**Comment by SchwartzN** [comment]
#

> I can confirm that something has changed. Our hooks added to "acf/init" are not working anymore, at all.
> 
> This is a bug? Or an intended, but undocumented change?!

We have the same issue here, our function that registers block_type are never called by the hook on ACF/Inuit.
It was working fine on 6.0.7 but it's broken since 6.1

**Comment by saulirajala** [comment]
We also have the same issue. It seems that every `add_action( 'acf/init', function(){} )` -call in our theme does actually nothing, since ACF has already triggered `acf/init`-hook before theme executes `add_action()`. I think the reason is [this](https://github.com/AdvancedCustomFields/acf/commit/ceb6559743f920a375572efffa122f61480aea08#diff-0a57e2acc62503a5eb744c29feab6784a986401a2b6bed1e4da63660e80f2a07R183) change in `acf_add_local_field_group()` -function in 6.1.1 update.

Depending on how and where you run `acf_add_local_field_group()` -functions, this can create race-condition situation where `acf_add_local_field_group()` (and thus `acf_init()` and thus `do_action('acf/init')` ) is executed before `add_action( 'acf/init', function(){} )` is executed in theme.

For example, we had following code in mu-plugin:
```php
add_action( 'plugins_loaded', function () {
	if ( function_exists( 'acf_add_local_field_group' ) ) :
		acf_add_local_field_group( $args );
	endif;
} );
```

When we changed it to use `acf/init` instead of `plugins_loaded` -hook, everything seems to be working fine.

However, the same issue occurs also, if I follow the [official documentation](https://www.advancedcustomfields.com/resources/register-fields-via-php/#basic) and add `acf_add_local_field_group()` straight to the `functions.php` of theme. So I'm not sure whether or not this is issue with ACF or just how we use ACF 🤔  Nevertheless at least the docs should be changed so that `acf_add_local_field_group()` is always inside of `acf/init` -hook in every code examples.

**Comment by saulirajala** [comment]
Nice work, 6.1.2 update seems to fix the issue in our case 🚀

**Comment by 3ele-projects** [comment]
Thank you very much.

I.e. you recognise it as a bug and I don't have to change my code and should wait for the update?

**Comment by mattgrshaw** [comment]
Hey folks, thanks for the reports on this one.

As mentioned earlier in the thread we released ACF 6.1.2 today, which should resolve most if not all of these issues.

We do still recommend registering things like field groups and options pages inside the `acf/init` hook if possible, so we're going to update our documentation and the ACF PHP export code to reflect this.
